store/cachekv/store.go

```diff
@@ -93,6 +93,28 @@ func (store *Store) Delete(key []byte) {
 	store.setCacheValue(key, nil, true)
 }
 
+func (store *Store) resetCaches() {
+	if len(store.cache) > 100_000 {
```

Comment:
In light of https://github.com/cosmos/cosmos-sdk/pull/15767#discussion_r1161558505 consider making this branch unconditional, that is, never use the cache clearing idiom.

The clearing was introduced as part of https://github.com/cosmos/cosmos-sdk/commit/dbb9923917be0e0c3d8a2b55c7cdd89428db998d, so I wonder how much performance is gained by the clearing itself. It not much, that's another reason to stop clearing.